app/src/features/sessionBook/screens/DraftSessionScreen/hooks/useSaveDraftSession.ts

```diff
@@ -0,0 +1,140 @@
+import { useIncentiveActions } from "features/incentivization/hooks";
+import { getRecordingOptionsToSave } from "features/sessionBook/utils/sessionFile";
+import { useCallback } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import { getCurrentlyActiveWorkspace } from "store/features/teams/selectors";
+import { getAppMode, getUserAttributes, getUserAuthDetails } from "store/selectors";
+import { CONSTANTS as GLOBAL_CONSTANTS } from "@requestly/requestly-core";
+import { getSessionRecordingEvents, getSessionRecordingMetaData } from "store/features/session-recording/selectors";
+import { saveRecording } from "backend/sessionRecording/saveRecording";
+import { compressEvents, getSessionEventsToSave } from "features/sessionBook/utils/sessionEvents";
+import { trackDraftSessionSaved, trackDraftSessionSaveFailed } from "features/sessionBook/analytics";
+import { toast } from "utils/Toast";
+import { DebugInfo, SessionSaveMode } from "features/sessionBook/types";
+import { IncentivizeEvent } from "features/incentivization/types";
+import { incentivizationActions } from "store/features/incentivization/slice";
+import { IncentivizationModal } from "store/features/incentivization/types";
+import PATHS from "config/constants/sub/paths";
+import { getAppFlavour } from "utils/AppUtils";
+
+export const useSaveDraftSession = () => {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const appMode = useSelector(getAppMode);
+  const user = useSelector(getUserAuthDetails);
+  const userAttributes = useSelector(getUserAttributes);
+  const currentlyActiveWorkspace = useSelector(getCurrentlyActiveWorkspace);
+  const { claimIncentiveRewards } = useIncentiveActions();
+  const sessionRecordingMetadata = useSelector(getSessionRecordingMetaData);
+  const sessionEvents = useSelector(getSessionRecordingEvents);
+
+  const saveDraftSessionHandler = useCallback(
+    async (recordingOptions: DebugInfo[], isOpenedInIframe: boolean, source: string) => {
+      const isDraftSession =
+        window.location.pathname.includes("draft") || appMode === GLOBAL_CONSTANTS.APP_MODES.DESKTOP;
+
+      const recordingOptionsToSave = getRecordingOptionsToSave(recordingOptions);
+      const appFlavour = getAppFlavour();
+      const clientSource =
+        appFlavour === GLOBAL_CONSTANTS.APP_FLAVOURS.REQUESTLY
+          ? GLOBAL_CONSTANTS.CLIENT_SOURCE.REQUESTLY
+          : GLOBAL_CONSTANTS.CLIENT_SOURCE.SESSIONBEAR;
+
+      if (isOpenedInIframe) {
+        window.parent.postMessage(
+          {
+            action: "draftSessionSaveClicked",
+            source: clientSource,
+            payload: {
+              name: sessionRecordingMetadata?.name,
+              description: sessionRecordingMetadata?.description,
+              recordingOptions: recordingOptions,
+            },
+          },
+          "*"
```

Comment:
Can we origin specific origin instead of asterisk for all the postMessages originating from the iframe and webApp?

Reply:
We cannot read the parent window's origin from the app loaded inside iframe

Reply:
Can you try using `document.referrer`?